src/amo/components/SectionLinks/index.js

```diff
@@ -2,40 +2,71 @@ import classNames from 'classnames';
 import React from 'react';
 import { compose } from 'redux';
 import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
 
 import Link from 'amo/components/Link';
 import type { ViewContextType } from 'amo/reducers/viewContext';
+import { setClientApp } from 'core/actions';
 import {
   ADDON_TYPE_EXTENSION,
   ADDON_TYPE_THEME,
+  CLIENT_APP_ANDROID,
+  CLIENT_APP_FIREFOX,
   VIEW_CONTEXT_EXPLORE,
   VIEW_CONTEXT_HOME,
+  VIEW_CONTEXT_LANGUAGE_TOOLS,
 } from 'core/constants';
+import type { ApiStateType } from 'core/reducers/api';
 import translate from 'core/i18n/translate';
 import { visibleAddonType } from 'core/utils';
+import DropdownMenu from 'ui/components/DropdownMenu';
+import DropdownMenuItem from 'ui/components/DropdownMenuItem';
 
 import './styles.scss';
 
 
 type SectionLinksProps = {
   className?: string,
+  clientApp: string,
+  dispatch: Function,
   i18n: Object,
+  router: Object,
   viewContext: ViewContextType,
 }
 
 export class SectionLinksBase extends React.Component {
+  setClientApp = (event) => {
```

Comment:
Could be nice to have a type for this event so that we know where it comes from (or at least have a hint).

Reply:
I tried using `Event` and `SyntheticEvent` but they complain about our (reasonable) use of `currentTarget` (and care about `target`) too.

Reply:
Yeah, this is a mess, and this reply does not really help: https://github.com/facebook/flow/issues/4381.